src/kv/mod.rs

```diff
@@ -1,35 +1,152 @@
+            let mut processing_watch_req = false;
+            loop {
+                if !processing_watch_req {
+                    if let Ok(watch_req) = watch_req_receiver.try_recv() {
+                        processing_key = watch_req.get_key();
+                        client_req_sender
+                            .send((watch_req.into(), WriteFlags::default()))
+                            .await
+                            .unwrap_or_else(|e| {
+                                panic!("Fail to send watch request, the error is: {}", e)
+                            });
+                        processing_watch_req = true;
+                    }
+                }
+                match client_resp_receiver.next().await {
+                    Some(watch_resp) => {
+                        match watch_resp {
+                            Ok(resp) => {
+                                if resp.get_created() {
+                                    cache_inner.insert_watch_id(
+                                        processing_key.clone(),
+                                        resp.get_watch_id(),
+                                    );
+                                    processing_watch_req = false;
+                                    continue;
+                                }
+                                if resp.get_canceled() {
+                                    cache_inner.delete_watch_id(&processing_key.clone());
+                                    processing_watch_req = false;
+                                    continue;
+                                }
+                                let events = resp.get_events().to_vec();
+                                for event in events {
+                                    if event.get_field_type() == Event_EventType::PUT {
+                                        match cache_inner.search(event.get_kv().get_key().to_vec())
+                                        {
+                                            Some(valid_value) => {
+                                                if valid_value.version
+                                                    >= event.get_kv().get_version()
+                                                {
+                                                    continue;
+                                                }
+                                            }
+                                            // If not in cache, ignore this value.
+                                            None => continue,
+                                        }
+                                        cache_inner.insert(
+                                            event.get_kv().get_key().to_vec(),
+                                            event.get_kv().clone(),
+                                        );
+                                    } else {
+                                        cache_inner.delete(event.get_kv().get_key().to_vec()).await;
```

Comment:
See if we need to cancel watch right here, since the key is already deleted in etcd server.